src/Microsoft.DotNet.CodeFormatting/Analyzers/UnwrittenWritableFieldFixer.cs

```diff
@@ -0,0 +1,59 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Collections.Immutable;
+using System.Diagnostics;
+using System.Linq;
+using System.Threading.Tasks;
+
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Text;
+using Microsoft.CodeAnalysis.CSharp;
+
+namespace Microsoft.DotNet.CodeFormatting.Analyzers
+{
+    [ExportCodeFixProvider(LanguageNames.CSharp)]
+    public class UnwrittenWritableFieldThisFixer : CodeFixProvider
+    {
+        private static SyntaxToken s_readOnlyToken =  SyntaxFactory.Token(SyntaxKind.ReadOnlyKeyword);
+
+        public override async Task RegisterCodeFixesAsync(CodeFixContext context)
+        {
+            SyntaxNode root = await context.Document.GetSyntaxRootAsync(context.CancellationToken).ConfigureAwait(false);
+            Diagnostic diagnostic = context.Diagnostics.First();
+            TextSpan diagnosticSpan = diagnostic.Location.SourceSpan;
+            var fieldDeclarationNode = root
+                .FindToken(diagnosticSpan.Start)
+                .Parent
+                .FirstAncestorOrSelf<FieldDeclarationSyntax>();
+
+            Debug.Assert(fieldDeclarationNode != null);
+
+            context.RegisterCodeFix(
+                CodeAction.Create(
+                    Resources.UnwrittenWritableFieldFixer_Title,
+                    c => AddReadonlyModifier(context.Document, root, fieldDeclarationNode)),
+                diagnostic);
+        }
+
+        private Task<Document> AddReadonlyModifier(Document document, SyntaxNode root, FieldDeclarationSyntax fieldDeclaration)
+        {
+            FieldDeclarationSyntax newFieldDeclaration = fieldDeclaration
+                .WithModifiers(fieldDeclaration.Modifiers.Add(s_readOnlyToken));
```

Comment:
Do we have a test to see what happens if the field is prefixed with a comment?  Consider:

``` csharp
/// <summary> stuff </summary>
int m_field;
```

In the past the `WithModifiers` method has gotten me in trouble because it would put the modifier before the comment instead of after which is the more logical place.  I believe I got into the most troubleu here with methods though so the same problem may not apply to fields.  And maybe it is limited to non-XML comments but either way just want to check if we have a test here.